qiskit/pulse/transforms/frames.py

```diff
@@ -0,0 +1,135 @@
+            a dict which can be used to initialized a ResolvedFrame.
+
+    Returns:
+        new_schedule: A new schedule where frames have been replaced with
+            their corresponding Drive, Control, and/or Measure channels.
+
+    Raises:
+        PulseError: if a frame is not configured.
+    """
+    if frames_config is None:
+        return schedule
+
+    resolved_frames = {}
+    sample_duration = None
+    for frame_settings in frames_config.values():
+        frame = ResolvedFrame(**frame_settings)
+        frame.set_frame_instructions(schedule)
+        resolved_frames[frame.index] = frame
+        sample_duration = frame_settings['sample_duration']
+
+    if sample_duration is None:
+        raise PulseError('Frame configuration does not have a sample duration.')
+
+    # Used to keep track of the frequency and phase of the channels
+    channel_trackers = {}
+    for ch in schedule.channels:
+        if isinstance(ch, chans.PulseChannel):
+            channel_trackers[ch] = ChannelTracker(ch, sample_duration)
+
+    # Add the channels that the frames broadcast on.
+    for frame in resolved_frames.values():
+        for ch in frame.channels:
+            if ch not in channel_trackers:
+                channel_trackers[ch] = ChannelTracker(ch, sample_duration)
+
+    sched = Schedule(name=schedule.name, metadata=schedule.metadata)
+
+    for time, inst in schedule.instructions:
+        if isinstance(inst, instructions.Play):
+            chan = inst.channel
+
+            if isinstance(inst.operands[0], Signal):
+                frame_idx = inst.operands[0].frame.index
+
+                if frame_idx not in resolved_frames:
+                    raise PulseError(f'{Frame(frame.index)} is not configured and cannot '
+                                     f'be resolved.')
+
+                frame = resolved_frames[frame_idx]
+
+                frame_freq = frame.frequency(time)
+                frame_phase = frame.phase(time)
+
+                # If the frequency and phase of the channel has already been set once in
+                # The past we compute shifts.
+                if channel_trackers[chan].is_initialized():
+                    freq_diff = frame_freq - channel_trackers[chan].frequency(time)
+                    phase_diff = frame_phase - channel_trackers[chan].phase(time)
+
+                    if freq_diff != 0.0:
+                        shift_freq = instructions.ShiftFrequency(freq_diff, chan)
+                        sched.insert(time, shift_freq, inplace=True)
+
+                    if phase_diff != 0.0:
+                        sched.insert(time, instructions.ShiftPhase(phase_diff, chan), inplace=True)
+
+                # If the channel's phase and frequency has not been set in the past
+                # we set it now
+                else:
+                    sched.insert(time, instructions.SetFrequency(frame_freq, chan), inplace=True)
```

Comment:
That will work and is the point of frames. The simplest example is a qutrit with 0-1 at 5.5 GHz and 1-2 at 5.2 GHz. Consider this example (which does something like `Xp`, `Xp12`, `Yp`)
```python
with pulse.build() as sched:
    with pulse.align_left():
        pulse.play(Signal(Gaussian(160, 0.144, 40), Frame(0)), DriveChannel(3))
        pulse.play(Signal(Gaussian(160, 0.1, 40), Frame(1)), DriveChannel(3))
        pulse.shift_phase(1.57, Frame(0))
        pulse.play(Signal(Gaussian(160, 0.144, 40), Frame(0)), DriveChannel(3))
```
This will be resolved to
```python
0, SetFrequency(5500000000.0, DriveChannel(3))
0, SetPhase(0.0, DriveChannel(3))
0, Play(Gaussian(duration=160, amp=(0.144+0j), sigma=40), DriveChannel(3))
160, SetFrequency(5200000000.0, DriveChannel(3))
160, SetPhase(4.4233624562545515, DriveChannel(3))
160, Play(Gaussian(duration=160, amp=(0.1+0j), sigma=40), DriveChannel(3))
320, SetFrequency(5500000000.0, DriveChannel(3))
320, SetPhase(6.093893421169632, DriveChannel(3))
320, Play(Gaussian(duration=160, amp=(0.144+0j), sigma=40), DriveChannel(3))
```
This will also work for CR.